cube/src/main/java/zone/cogni/asquare/cube/convertor/data2shacl/ShaclGenerator.java

```diff
@@ -458,12 +481,17 @@ private List<String> selectForTypeAndProperty(RdfStoreService rdfStoreService,
                                                 String fileName,
                                                 Resource type,
                                                 Resource property) {
+    List<Map<String, RDFNode>> rows = getRowsForTypeAndProperty(rdfStoreService, fileName, type, property);
+    return paginatedQuery.convertSingleColumnUriToStringList(rows);
+  }
+
+  private List<Map<String, RDFNode>> getRowsForTypeAndProperty(RdfStoreService rdfStoreService, String fileName, Resource type, Resource property) {
```

Comment:
can this be split in a getTypeAndProperty params which returns the parameter map 
then a getRows(RdfStoreService rdfStoreService, String fileName,Map<String, String> parameters) method